### PR TITLE
[Core] empty content fix

### PR DESCRIPTION
**Pull request by Stevenjin8**
# Description

Before, `HttpRequest.content` would return `None` if we passed in `content=b""` when initializing the request object. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Azure-core is supported on Python 3.7 or later. For more details, please read ou
 
 - Fixed the bug that azure-core could not be imported under Python 3.11.0b3  #24928
 - `ContentDecodePolicy` can now correctly deserialize more JSON bodies with different mime types #22410
+- Fixed the bug where `HttpRequest.content` would return None when initialized with `content=b""` #25489
```
Comment (Stevenjin8) [inline sdk/core/azure-core/CHANGELOG.md:17]:
```suggestion
- Fixed the bug where `HttpRequest.content` would return `None` when initialized with `content=b""`#25489
```

**Review by iscai-msft — APPROVED** [review]
thanks @Stevenjin8

**Review by lmazuel — CHANGES_REQUESTED** [review]
I'm not sure we want to make a semantic difference between `None` and empty body, those are technically the same thing in the HTTP spec. If we go that path. more trouble may come.
I'm not against the PR, but we shouldn't let the fast that this is a short PR let us think this is an easy problem, this should be discussed further.

**Comment by johanste** [comment]
status code and content-type play into this. Status code 204, no content. Status 200, content-type application/octet-stream, content-length 0 is an empty `bytes`. Many azure services get this wrong, though.

**Comment by Stevenjin8** [comment]
I should have added more context to this PR. This PR is a fix for #25489 where a customer was having issues uploading empty blobs to container storage. The issue arises when trying to computer the MD5 hash of a request with an empty body:

https://github.com/Azure/azure-sdk-for-python/blob/50d8cab2e7e6a57592e864b14798a5e2aa7306d4/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py#L352-L371

Since `None` is not a readable file nor a bytes object, this above method raises a value error. An alternative method to doing this would be to update the `get_content_md5` method to have something along the lines of

```python
data = data or b""
```

**Comment by Stevenjin8** [comment]
@johanste 

> status code and content-type play into this. Status code 204, no content. Status 200, content-type application/octet-stream, content-length 0 is an empty `bytes`. Many azure services get this wrong, though.

@johanste Not sure how status code plays into this since this PR is about requests, not responses.

**Comment by Stevenjin8** [comment]
#25489 was resolved by #25655, so this PR is less of a priority, but still worth considering

**Pull request closed without merging**